src/Concordium/Client/Runner.hs

```diff
@@ -1494,6 +1485,22 @@ tailTransaction verbose hash = do
   where
     getLocalTimeOfDayFormatted = showTimeOfDay <$> getLocalTimeOfDay
 
+-- |`read` input or fail if the input could not be `read`.
+readOrFail :: (MonadIO m, Read a) => Text -> m a
+readOrFail t =
+  case readEither s of
+    Left err -> logFatal [[i|Unable to parse '#{s}': #{err}|]]
+    Right v -> return v
+  where s = Text.unpack t
+
+-- |Reads a blockhash wrapped in the `Maybe` monad.
+-- If the provided value is @Nothing@, a default value specified provided
+-- as the first parameter is returned. If the provided value is string wrapped
+-- in @Just@, `readOrFail s` is returned. Fails if `s` is not a valid blockhash.
```

Comment:
When you refer to `s` it forces the user of this function to visit the definition. Ideally, you would be able to just read the comment and understand how to use it. Especially for such a simple function.

Reply:
Also, as I commetned in the other PR where you have the same function, please don't use "monad" for this. There is nothing at all "monadic" in this use of Maybe, so it is just needles jargon.